addons/website_crm_partner_assign/controllers/main.py

```diff
@@ -100,14 +99,14 @@ def portal_my_opportunities(self, page=1, date_begin=None, date_end=None, sortby
         domain = self.get_domain_my_opp(request.env.user)
 
         today = fields.Date.today()
-        this_week_end_date = fields.Date.to_string(fields.Date.from_string(today) + datetime.timedelta(days=7))
+        user_activity_leads = CrmLead.search([('activity_user_id', '=', request.env.user.id)]).ids if filterby == 'no_activities' else []
 
         searchbar_filters = {
             'all': {'label': _('Active'), 'domain': []},
+            'no_activities': {'label': _('No Activities'), 'domain': [('id', 'not in', user_activity_leads)]},
+            'overdue': {'label': _('Late Activities'), 'domain': [('activity_date_deadline', '<', today)]},
             'today': {'label': _('Today Activities'), 'domain': [('activity_date_deadline', '=', today)]},
-            'week': {'label': _('This Week Activities'),
-                     'domain': [('activity_date_deadline', '>=', today), ('activity_date_deadline', '<=', this_week_end_date)]},
-            'overdue': {'label': _('Overdue Activities'), 'domain': [('activity_date_deadline', '<', today)]},
+            'future': {'label': _('Future Activities'), 'domain': [('activity_date_deadline', '>=', today)]},
```

Comment:
Hello @nihp-odoo  :smile: 
Just a small comment :slightly_smiling_face: 
Isn't it `>` instead of `>=` here ? :thinking: If you set it to `>=` you'll get today's activities too.

Reply:
Hello @amdi-odoo 
Yeah, but I added it just to be safe. What if the activity we had scheduled today is later maybe after 2 or 3hrs. So, just to cover such cases I did `>=` :smile:

Reply:
Yup @amdi-odoo you have a point. The changes are done. Just waiting for the runbot.. :hourglass_flowing_sand: 
Please have a look.
Thanks.. :smile: